src/mapping/NearestNeighborBaseMapping.cpp

```diff
@@ -61,9 +61,11 @@ void NearestNeighborBaseMapping::computeMapping()
 
   auto &index = searchSpace->index();
   for (size_t i = 0; i < verticesSize; ++i) {
-    const auto &matchedVertex = index.getClosestVertex(sourceVertices[i].getCoords());
+    const auto &sourceCoords  = sourceVertices[i].getCoords();
+    const auto &matchedVertex = index.getClosestVertex(sourceCoords);
     _vertexIndices[i]         = matchedVertex.index;
-    distanceStatistics(matchedVertex.distance);
+    auto distance             = (sourceCoords - searchSpace->vertices()[matchedVertex.index].getCoords()).norm();
```

Comment:
We should add a comment why we compute the distance here manually instead of using the boost functionality. Maybe even link to the issue.

Reply:
Actually, we could still use the distance of boost geometry here.
My long term goal is to limit boost geometry to the query package. These headers are expensive.

Reply:
I thought the bug is about the distance calculation? If that's not the case here, we could still rely on boost. I guess with 'expensive' you refer to compile time? How does it look like in terms of runtime performance (if you really want to open these boxes now).

Reply:
> I thought the bug is about the distance calculation?

It is. I removed the distances from the `getClosestX` results. So, one still needs to compute the distance here.

> If that's not the case here, we could still rely on boost.

I disagree. We use Eigen for coordinates and data, so we should use it here as well.
Boost.geometry and all the boilerplate that comes with it is purely useful for the spacial index and geometric queries (like distances to AABBs etc). 

> I guess with 'expensive' you refer to compile time?

Exactly. Eigen is expensive enough on its own. We should pull in another dependency for something that Eigen was build for.

Reply:
I clarified the code a bit.